lib/components/joystick/joystick_action.dart

```diff
@@ -149,13 +137,15 @@ class JoystickAction {
       final double nextY = dist * sin(_radAngle);
       final Offset nextPoint = Offset(nextX, nextY);
 
-      final Offset diff = Offset(
-            _rectBackgroundDirection.center.dx + nextPoint.dx,
-            _rectBackgroundDirection.center.dy + nextPoint.dy,
-          ) -
-          _rectAction.center;
+      if (_rectAction != null) {
+        final Offset diff = Offset(
+              _rectBackgroundDirection.center.dx + nextPoint.dx,
+              _rectBackgroundDirection.center.dy + nextPoint.dy,
+            ) -
+            _rectAction.center;
```

Comment:
existing code, but this could be:

```dart
final nextPoint = Vector2(nextX, nextY);
// ...
final Offset diff = _rectBackgroundDirection.center.toVector2() + nextPoint - _rectAction.center;
_rectAction = _rectAction.shift(diff.toOffset());
```

Reply:
Let's change this afterward. Created #606.